### PR TITLE
ENT-4432: Quote package module path

### PR DESCRIPTION
**Pull request by vpodzime**
See commit messages for details.

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2026/)

https://ci.cfengine.com/job/pr-pipeline/2026/

**Comment by vpodzime** [comment]
A new jenkins build:

**Comment by vpodzime** [comment]
> @vpodzime vpodzime force-pushed the...

**Comment by vpodzime** [comment]
A new builds (with exotics):
[Build Status](https://ci.cfengine.com/job/pr-pipeline/2032/)

**Comment by vpodzime** [comment]
> A new build (with exotics):
> [Build Status](https://ci.cfengine.com/job/pr-pipeline/2032/)

Solaris 10 got lost in the clouds, but all green otherwise.

File: libutils/path.c [hunk 1]
```diff
@@ -32,3 +33,36 @@ char *Path_JoinAlloc(const char *dir, const char *leaf)
     }
     return StringConcatenate(3, dir, "/", leaf);
 }
+
+char *Path_GetQuoted(const char *path)
+{
+    if (path == NULL)
+        return NULL;
```
Comment (olehermanse) [inline libutils/path.c]: Our style says always curlys.

File: libutils/path.c [hunk 2]
```diff
@@ -32,3 +33,36 @@ char *Path_JoinAlloc(const char *dir, const char *leaf)
...
+    bool needs_quoting = false;
+    for (const char *cp = path; !needs_quoting && (*cp != '\0'); cp++)
+    {
+        /* let's quote everything that's not just alphanumerics, underscores and
+         * dashes */
+        needs_quoting = !(((*cp >= 'a') && (*cp <= 'z')) ||
+                          ((*cp >= 'A') && (*cp <= 'Z')) ||
+                          ((*cp >= '0') && (*cp <= '9')) ||
+                          (*cp == '_') || (*cp == '-')   ||
+                          IsFileSep(*cp));
```
Comment (olehermanse) [inline libutils/path.c]: Nice

File: tests/unit/path_test.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    assert_string_equal(result, "\"needs some/quoting\"");
+    free(result);
+
+    result = Path_GetQuoted("also&needs/quoting");
+    assert_string_equal(result, "\"also&needs/quoting\"");
+    free(result);
+
+    result = Path_GetQuoted("also;needs/quoting");
+    assert_string_equal(result, "\"also;needs/quoting\"");
+    free(result);
```
Comment (olehermanse) [inline tests/unit/path_test.c]: 👏

File: libutils/path.c (hunk 1 above)
Comment (vpodzime) [reply libutils/path.c]: Done!

**Pull request merged**